compiler/rustc_typeck/src/collect.rs

```diff
@@ -1986,6 +1986,17 @@ fn predicates_defined_on(tcx: TyCtxt<'_>, def_id: DefId) -> ty::GenericPredicate
 fn predicates_of(tcx: TyCtxt<'_>, def_id: DefId) -> ty::GenericPredicates<'_> {
     let mut result = tcx.predicates_defined_on(def_id);
 
+    if tcx.has_attr(def_id, sym::default_method_body_is_const) {
+        if let Some(parent) = tcx.parent(def_id).filter(|&did| tcx.is_trait(did)) {
```

Comment:
I think you can assert this instead of filtering. We only allow the attribute in traits

Reply:
I added this because it was ICEing and I had to guard against non-trait-method usages. For some reason the wrong attribute usage doesn't stop compilation and continues to typeck ...

Reply:
Yea... I feared that that was the issue. OK then, let's leave it